crates/virtio-queue/src/mock.rs

```diff
@@ -308,13 +314,15 @@ impl<'a, M: GuestMemory> MockSplitQueue<'a, M> {
         self.used_addr
     }
 
-    fn update_avail_idx(&mut self, value: u16) {
+    fn update_avail_idx(&mut self, value: u16) -> Result<(), Error> {
         let avail_idx = self.avail.idx.load();
-        self.avail.ring.ref_at(avail_idx as usize).store(value);
+        let avail_ref = self.avail.ring.ref_at(avail_idx as usize)?;
+        avail_ref.store(value);
```

Comment:
nit: you don't need to split this into two, you can also use:
`self.avail.ring.ref_at(avail_idx as usize)?.store(value);`